design/batch_retrieval_design.md

```diff
@@ -1,20 +1,354 @@
+| 1    | Vanilla flow                                                 | Given I am an authenticated, privileged user/host<br />With proper permissions on requested secrets | I send `POST /secrets` with valid JSON body with  `var/path/conjurSecret` and `var/path/anotherConjurSecret` in request body | - The JSON response should be (*see below* #1)<br />- Status response code should be `200 OK`<br />- The proper audit record should appear for the whole request and each individual secret | Automated        |
+| 2    | User/Host authenticated but does not have `execute` privilege | Given I am a privileged, authenticated user/host<br />With proper permissions on ***one*** of requested secrets | I send `POST /secrets` with valid JSON body with  `var/path/conjurSecret` and `var/path/anotherConjurSecret` in request body | - The JSON response should be (*see below* #2)<br />- Status response code should be `200 OK`<br />- Existing secret should have `200`. Non-existing secret should have 404 status error<br />- The proper audit record should appear for the whole request and each individual secret | Automated        |
+| 3    | User/Host authenticated but secret doesn't exist             | Given I am a privileged, authenticated user/host<br />With proper `execute` permissions<br />Requesting secret does not exist | I send `POST /secrets` with valid JSON body with  `var/path/conjurSecret` and `var/path/nonExistingSecret` in request body | - The JSON response should be (*see below #3*)<br />- Status response code should be `200 OK`<br />- Existing secret should have `200`. Non-existing secret should have `404 ` status error<br />- The proper audit record should appear for the whole request and each individual secret | Automated        |
+| 4    | User/Host authenticated but secret does not have value       | Given I am a privileged, authenticated user/host<br />With proper permissions<br />Requesting secret *value* does not exist | I send `POST /secrets` with valid JSON body with  `var/path/conjurSecret` and `var/path/nonSecretValue` in request body | - The JSON response should be (*see below #4)<br />- Status response code should be `200 OK`<br />- Existing secret should have `200`. Non-existing secret value should have `404 ` status error<br />- The proper audit record should appear for the whole request and each individual secret | Automated        |
+| 5    | Invalid request body                                         | Given I am a privileged, authenticated user/host<br />With proper permissions<br />Requesting secret in invalid format | I send `POST /secrets` with invalid JSON body with `var/path/conjurSecret` and  `var/path/<>` | - The JSON response should be (see below #5)<br />- Status response code should be `400 `<br />- The proper audit record should appear for the whole request and each individual secret<br />- **@Elad should the whole request fail?** | Automated        |
```

Comment:
Suppose to fail since we have problem with the request

Reply:
agree

Reply:
fail the _whole_ request? Adding #5 (: